data_transfer/devices/dreem.py

```diff
@@ -88,53 +95,37 @@ def download_metadata(self) -> None:
                 )
             )
 
-            # Reformat to mirror UCAM/DMP.
-            patient_id = patient_id.replace("-", "") if patient_id else patient_id
-
-            if patient_id and (ucam_entry := ucam.get_record(patient_id)):
-                known += 1
-                dreem_devices = [
-                    d for d in ucam_entry.devices if self.device_type in d.device_id
-                ]
-
-                # Best-case: only one device was worn and UCAM knows it
-                if len(dreem_devices) == 1:
-                    device_record = dreem_devices[0]
-                # Edge-case: multiple dreem headbands used, e.g., if one broke.
-                elif len(dreem_devices) > 1:
-                    # Determine usage based on weartime as it exists in UCAM
-                    device_record = ucam.record_by_wear_period_in_list(
-                        dreem_devices, recording.start, recording.end
-                    )
-                # Edge-case: device not logged with patient in UCAM
-                else:
-                    print(f"Metadata cannot be determined for:\n    {recording}")
-                    continue
-            else:
+            device_id = self.__device_id_from_ucam(
+                patient_id, recording.start, recording.end
+            ) or inventory.device_id_by_serial(device_serial)
+
+            if not patient_id or not device_id:
+                log.debug(f"Unknown Device:\n   {recording}")
+                log.debug(f"Patient ({patient_id}) or Device ID ({device_id}) missing.")
                 unknown += 1
-                print(f"Metadata cannot be determined for:\n    {recording}")
-                continue
+                continue  # Skip record
+            known += 1
 
             record = Record(
-                filename=recording.id,
+                hash=recording.id,
+                manufacturer_ref=recording.id,
                 device_type=self.device_type,
                 patient_id=patient_id,
-                **asdict(device_record),
+                device_id=device_id,
+                start_wear=recording.start,
+                end_wear=recording.end,
             )
 
             create_record(record)
-            print(f"Record Created:\n   {record}")
 
-            path = Path(config.storage_vol / f"{record.filename}-meta.json")
             # Store metadata from memory to file
-            utils.write_json(path, item)
+            utils.write_json(record.metadata_path(), item)
```

Comment:
When you have the added hash, and following the BTF implementation, this line should be preceded by:
```python
del item["IDEAFAST"]  # remove record-specific temporary metadata
```

Reply:
@jawrainey , This removes all metadata generated temporarily for the pipeline